daemon/runtime_unix.go

```diff
@@ -85,7 +85,9 @@ func (daemon *Daemon) initRuntimes(runtimes map[string]types.Runtime) (err error
 		}
 
 		if err = os.Rename(runtimeDir, runtimeDir+"-old"); err != nil {
-			return
+			if err = os.RemoveAll(runtimeDir); err != nil {
```

Comment:
Just purging it completely feels a bit drastic (although I admit I am not familiar with why we rename to `-old` to begin with).

Do you think this warrants at least a `logrus.WithError(err).Warn(...)` similar to the below cases so we can at least tell from the logs that a fallback from `Rename` to `RemoveAll` happened in case there are other cases later beyond today's `overlayfs` one that happen to hit this fallback?

Reply:
> Just purging it completely feels a bit drastic

We are doing the same few lines below right after renaming `tmpDir`to `runtimeDir`.

> I admit I am not familiar with why we rename to -old to begin with

I'm also not familiar with this logic, but seems it's just a some kind of `backup` in case we failed, but anyway only renaming `tmpDir` to `runtimeDir` is covered by this "backup". So, not sure how useful it is.

I added logging and some minor changes. Please let me know if you think I can improve something here.

Now it looks like this:
```
WARN[2022-02-13T22:54:54.834557200Z] failed to rename runtimes dir to old. Will try to removing it  dir=/docker/runtimes error="rename /docker/runtimes /docker/runtimes-old: invalid cross-device link"

```

Reply:
> a quick glance showed that there's a os.Rename() on line 97 as well; does that one not have the same issue? (if rename is not supported)?

@thaJeztah as you can see per [comment](https://github.com/moby/moby/issues/25409#issuecomment-239756567) `os.rename` doesn't work in case of renaming file/directory which was existing in lower/merged.

As `tmpDir` was created few lines before on upper FS, it does not occur any issues.